lib/autoload/course/conditional/user_satisfiability_graph.rb

```diff
@@ -59,7 +59,7 @@ def evaluate(satisfied_conditions, course_user)
   def self.reachable?(source, dest)
     return true if source == dest
 
-    dest.specific_conditions.index { |c| reachable?(source, c.dependent_object) }.present?
+    dest.specific_conditions.index { |c| reachable?(source, c.dependent_object) }.present? if dest
```

Comment:
this would return nil. I suggest you explicitly return false unless dest at the start of the function